datapipelines/src/test/java/io/customer/datapipelines/DataPipelinesInteractionTests.kt

```diff
@@ -54,4 +70,99 @@ class DataPipelinesInteractionTests : UnitTest() {
             .shouldNotBeNull()
             .shouldMatchTo(givenTraits)
     }
+
+    @Test
+    fun identify_givenIdentifierWithJson_expectSetNewProfileWithAttributes() {
+        val givenIdentifier = String.random
+        val givenFirstNamePair = "first_name" to "Dana"
+        val givenAgePair = "ageInYears" to 30
+        val givenTraits = mapOf(givenFirstNamePair, givenAgePair)
+
+        analytics.userId().shouldBeNull()
+
+        sdkInstance.identify(
+            givenIdentifier,
+            buildJsonObject {
+                put(givenFirstNamePair.first, givenFirstNamePair.second)
+                put(givenAgePair.first, givenAgePair.second)
+            }
+        )
+
+        analytics.userId() shouldBeEqualTo (givenIdentifier)
+        analytics.traits().shouldNotBeNull() shouldMatchTo givenTraits
+
+        outputReaderPlugin.identifyEvents.size shouldBeEqualTo 1
+        val identifyEvent = outputReaderPlugin.identifyEvents.lastOrNull()
+        identifyEvent.shouldNotBeNull()
+
+        identifyEvent.userId shouldBeEqualTo givenIdentifier
+        identifyEvent.traits.shouldNotBeNull() shouldMatchTo givenTraits
+    }
+
+    @Test
+    fun identify_givenIdentifierWithTraits_expectSetNewProfileWithAttributes() {
+        val givenIdentifier = String.random
+        val givenTraits = UserTraits(
+            firstName = "Dana",
+            ageInYears = 30
+        )
+        val givenTraitsJson = givenTraits.encodeToJsonElement()
+
+        analytics.userId().shouldBeNull()
+
+        sdkInstance.identify(givenIdentifier, givenTraits)
+
+        analytics.userId() shouldBeEqualTo givenIdentifier
+        analytics.traits().shouldNotBeNull() shouldBeEqualTo givenTraitsJson
+
+        outputReaderPlugin.identifyEvents.size shouldBeEqualTo 1
+        val identifyEvent = outputReaderPlugin.identifyEvents.lastOrNull()
+        identifyEvent.shouldNotBeNull()
+
+        identifyEvent.userId shouldBeEqualTo givenIdentifier
+        identifyEvent.traits.shouldNotBeNull() shouldBeEqualTo givenTraitsJson
+    }
+
+    @Test
+    fun identify_givenEmptyIdentifier_givenNoProfilePreviouslyIdentified_expectRequestIgnored() {
+        val givenIdentifier = ""
+
+        sdkInstance.identify(givenIdentifier)
+
+        analytics.userId().shouldBeNull()
+        analytics.traits() shouldBeEqualTo emptyJsonObject
+
+        outputReaderPlugin.allEvents.shouldBeEmpty()
+    }
+
+    @Test
+    fun identify_givenEmptyIdentifier_givenProfileAlreadyIdentified_expectRequestIgnored() {
```

Comment:
what happens if we previously identified a profile with traits, and then identified with empty identifier and empty traits, can we add a test for it?

Reply:
Sure we can, added.